lib/puppet/resource/type_collection.rb

```diff
@@ -187,7 +187,7 @@ def version
       if environment.config_version.nil? || environment.config_version == ""
         @version = Time.now.to_i
       else
-        @version = Puppet::Util::Execution.execute([environment.config_version]).strip
+        @version = Puppet::Util::Execution.execute(environment.config_version).strip
```

Comment:
I'm not sure what `execve` is.

According to the yard doc for Puppet::Util::Execution.execute, the first argument should only be an array if the first element of the array is the executable, and should be a string otherwise.